nanofi/src/core/flowfiles.c

```diff
@@ -0,0 +1,99 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "api/nanofi.h"
+#include "core/flowfiles.h"
+
+#include "utlist.h"
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+void add_flow_file_record(flow_file_list ** ff_list, flow_file_record * record) {
+    if (!record) return;
+
+    struct flow_file_list * new_node = (struct flow_file_list *)malloc(sizeof(struct flow_file_list));
+    new_node->ff_record = record;
+    LL_APPEND(*ff_list, new_node);
+}
+
+void free_flow_file_list(flow_file_list ** ff_list) {
+    if (!*ff_list) {
+        return;
+    }
+    flow_file_list * el = NULL;
+    LL_FOREACH(*ff_list, el) {
+        if (el) {
+            free_flowfile(el->ff_record);
+        }
+    }
+}
+
+flow_file_record * write_to_flow_file(const char * buff, size_t count, nifi_instance * instance, standalone_processor * proc) {
+    if (!buff || !instance || !proc) {
```

Comment:
The null check macro can still be used in such cases

Reply:
I think this is more readable than using the macro.